src/server/scripts/Zandalar/AtalDazar/boss_priestess_alun_za.cpp

```diff
@@ -0,0 +1,684 @@
+{
+    void OnRemove(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/) const
+    {
+        if (GetTargetApplication()->GetRemoveMode() != AURA_REMOVE_BY_EXPIRE)
+            return;
+
+        Unit* target = GetTarget();
+        if (!target)
+            return;
+
+        target->CastSpell(target, GetEffectInfo(EFFECT_0).CalcValue(), CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringAura = aurEff
+        });
+    }
+
+    void Register() override
+    {
+        OnEffectRemove += AuraEffectRemoveFn(spell_priestess_alun_za_molten_gold::OnRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 255584 - Molten Gold
+class spell_priestess_alun_za_molten_gold_selector : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_MOLTEN_GOLD_MISSILE });
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/) const
+    {
+        GetCaster()->CastSpell(GetHitUnit(), SPELL_MOLTEN_GOLD_MISSILE, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell(),
+            .OriginalCastId = GetSpell()->m_castId
+        });
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_priestess_alun_za_molten_gold_selector::HandleScript, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 255566 - Bubble
+class spell_priestess_alun_za_bubble : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_TAINTED_BLOOD_MISSILE });
+    }
+
+    void HandleScript(SpellEffIndex /*effIndex*/) const
+    {
+        GetHitUnit()->CastSpell(*GetHitDest(), SPELL_TAINTED_BLOOD_MISSILE, CastSpellExtraArgsInit{
+            .TriggerFlags = TRIGGERED_IGNORE_CAST_IN_PROGRESS | TRIGGERED_DONT_REPORT_CAST_ERROR,
+            .TriggeringSpell = GetSpell(),
+            .OriginalCastId = GetSpell()->m_castId
+        });
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_priestess_alun_za_bubble::HandleScript, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 255565 - Bubble
+class spell_priestess_alun_za_tainted_blood : public SpellScript
+{
+    static void SetDest(SpellDestination& dest)
+    {
+        dest.Relocate(Trinity::Containers::SelectRandomContainerElement(TaintedBloodPosition));
+    }
+
+    void Register() override
+    {
+        OnDestinationTargetSelect += SpellDestinationTargetSelectFn(spell_priestess_alun_za_tainted_blood::SetDest, EFFECT_0, TARGET_DEST_DEST);
+    }
+};
+
+// 255577 - Transfusion
+class spell_priestess_alun_za_tranfusion_cast : public SpellScript
+{
+    void HandleEnergy(SpellEffIndex /*effIndex*/) const
+    {
+        GetCaster()->SetPower(POWER_ENERGY, 0);
```

Comment:
Is this necessary? Spell 255577 has 100 energy power cost, this shouldn't be required.

Reply:
didnt realize that